src/librustc/dep_graph/graph.rs

```diff
@@ -476,10 +476,8 @@ impl DepGraph {
             fingerprints.resize(current_dep_graph.nodes.len(), Fingerprint::ZERO);
         }
 
-        let nodes: IndexVec<_, (DepNode, Fingerprint)> =
-            current_dep_graph.nodes.iter_enumerated().map(|(idx, &dep_node)| {
-            (dep_node, fingerprints[idx])
-        }).collect();
+        let fingerprints: IndexVec<_, _> = fingerprints.iter().cloned().collect();
+        let nodes : IndexVec<_, _> = current_dep_graph.nodes.iter().cloned().collect();
```

Comment:
We should just use `.clone()` on the `IndexVec`s directory. That will make it more likely that this is a simple `memcpy`.

Reply:
Good idea! I was so busy looking for `impl Clone for IndexVec` I totally missed the `derive(Clone)` line. 😆

Reply:
:smile: